### ISSUE TITLE
(18.03 & master) amarok-kf5 is broken.

### ISSUE THREAD

**Issue by cx405** [NixOS/nixpkgs#38026]
## Issue description
Application displays a dialog box with "Amarok could not find any plugins. This indicates an installation problem." and then quits.

Regular output to console:
```
$ amarok
**********************************************************************************************
** AMAROK WAS STARTED IN NORMAL MODE. IF YOU WANT TO SEE DEBUGGING INFORMATION, PLEASE USE: **
** amarok --debug                                                                           **
**********************************************************************************************
kf5.kservice.services: KServiceTypeTrader: serviceType "Amarok/Plugin" not found
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
kf5.kservice.services: KServiceTypeTrader: serviceType "Amarok/Plugin" not found
```

Full debug output:
```
$ PHONON_BACKEND_DEBUG=3 PHONON_SUBSYSTEM_DEBUG=1 amarok --debug
amarok: BEGIN: void App::continueInit()
amarok:   BEGIN: virtual int App::newInstance()
amarok:     BEGIN: void App::handleCliArgs(const QString&)
amarok:     END__: void App::handleCliArgs(const QString&) [Took: 0s]
amarok:   END__: virtual int App::newInstance() [Took: 0s]
amarok:   BEGIN: EngineController::EngineController()
amarok:   END__: EngineController::EngineController() [Took: 0.003s]
amarok:   BEGIN: void EngineController::initializePhonon()
PHONON-GST     Using GStreamer 1.12.3
PHONON-GST     AudioOutput using "pulsesink"
PHONON-GST     AudioOutput using "pulsesink"
PHONON-GST     Backend connected Phonon::Gstreamer::MediaObject to Phonon::Gstreamer::AudioOutput
amarok:     BEGIN: void EqualizerController::initialize(const Phonon::Path&)
amarok:       [EqualizerController] Established Phonon equalizer effect with 10 parameters.
amarok:       BEGIN: void EqualizerController::eqUpdate()
amarok:       END__: void EqualizerController::eqUpdate() [Took: 0.001s]
amarok:     END__: void EqualizerController::initialize(const Phonon::Path&) [Took: 0.003s]
PHONON-GST     Fading to 1
PHONON-GST     BEGIN: GstStateChangeReturn Phonon::Gstreamer::Pipeline::setState(GstState)
PHONON-GST       Transitioning to state "ready"
PHONON-GST       State change
PHONON-GST       BEGIN: void Phonon::Gstreamer::MediaObject::handleDurationChange(qint64)
PHONON-GST         -1
PHONON-GST       END__: void Phonon::Gstreamer::MediaObject::handleDurationChange(qint64) [Took: 0s]
PHONON-GST       BEGIN: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState)
PHONON-GST         Moving from "null"  Phonon::LoadingState  to "ready"  Phonon::StoppedState 
PHONON-GST         BEGIN: void Phonon::Gstreamer::MediaObject::loadingComplete()
PHONON-GST         END__: void Phonon::Gstreamer::MediaObject::loadingComplete() [Took: 0s]
PHONON-GST       END__: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState) [Took: 0s]                                                                                                       
PHONON-GST     END__: GstStateChangeReturn Phonon::Gstreamer::Pipeline::setState(GstState) [Took: 0s]
PHONON-GST     Backend connected Phonon::Gstreamer::MediaObject to Phonon::Gstreamer::VolumeFaderEffect
PHONON-GST     Backend connected Phonon::Gstreamer::VolumeFaderEffect to Phonon::Gstreamer::AudioOutput
PHONON-GST     Fading to 1
PHONON-GST     BEGIN: GstStateChangeReturn Phonon::Gstreamer::Pipeline::setState(GstState)
PHONON-GST       Transitioning to state "ready"
PHONON-GST     END__: GstStateChangeReturn Phonon::Gstreamer::Pipeline::setState(GstState) [Took: 0s]
PHONON-GST     Backend connected Phonon::Gstreamer::VolumeFaderEffect to Phonon::Gstreamer::VolumeFaderEffect
PHONON-GST     Backend connected Phonon::Gstreamer::VolumeFaderEffect to Phonon::Gstreamer::AudioOutput
PHONON-GST     Backend connected Phonon::Gstreamer::VolumeFaderEffect to Phonon::Gstreamer::AudioDataOutput
amarok:     [EngineController] Tick Interval (actual):  100
amarok:   END__: void EngineController::initializePhonon() [Took: 0.13s]
amarok:   BEGIN: CollectionManager::CollectionManager()
amarok:   END__: CollectionManager::CollectionManager() [Took: 0s]
amarok:   BEGIN: MainWindow::MainWindow()
amarok:     BEGIN: Plugins::PluginManager::PluginManager(QObject*)
kf5.kservice.services: KServiceTypeTrader: serviceType "Amarok/Plugin" not found
amarok:       [PluginManager] 0 plugins in total
amarok:       BEGIN: void Plugins::PluginManager::handleNoPluginsFound()
amarok:         [PluginManager] No Amarok plugins found, running kbuildsycoca4.
QThread::wait: Thread tried to wait on itself
amarok:         [EngineController] slotTrackLengthChanged( -1 )
amarok:         [EngineController] slotStateChanged from  Phonon::LoadingState  to  Phonon::StoppedState 
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
QThread::wait: Thread tried to wait on itself
amarok:         [PluginManager] Second attempt at finding plugins
kf5.kservice.services: KServiceTypeTrader: serviceType "Amarok/Plugin" not found
amarok:         [PluginManager] 0 plugins in total
PHONON-GST         AudioOutput using "pulsesink"
PHONON-GST         Backend connected Phonon::Gstreamer::MediaObject to Phonon::Gstreamer::AudioOutput
PHONON-GST         BEGIN: virtual void Phonon::Gstreamer::MediaObject::setSource(const Phonon::MediaSource&)
PHONON-GST           Setting new source
PHONON-GST           New source: QUrl("file:///run/current-system/sw/share/sounds/Oxygen-Sys-App-Error-Critical.ogg")
PHONON-GST           uri "file:///run/current-system/sw/share/sounds/Oxygen-Sys-App-Error-Critical.ogg"
PHONON-GST         END__: virtual void Phonon::Gstreamer::MediaObject::setSource(const Phonon::MediaSource&) [Took: 0s]
PHONON-GST         BEGIN: virtual void Phonon::Gstreamer::MediaObject::play()
PHONON-GST           BEGIN: void Phonon::Gstreamer::MediaObject::requestState(Phonon::State)
PHONON-GST              Phonon::PlayingState 
PHONON-GST             BEGIN: GstStateChangeReturn Phonon::Gstreamer::Pipeline::setState(GstState)
PHONON-GST               Transitioning to state "playing"
PHONON-GST               State change
PHONON-GST               BEGIN: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState)
PHONON-GST                 Moving from "null"  Phonon::LoadingState  to "ready"  Phonon::StoppedState 
PHONON-GST                 BEGIN: void Phonon::Gstreamer::MediaObject::loadingComplete()
PHONON-GST                 END__: void Phonon::Gstreamer::MediaObject::loadingComplete() [Took: 0s]
PHONON-GST               END__: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState) [Took: 0s]
PHONON-GST               BEGIN: static void Phonon::Gstreamer::Pipeline::cb_setupSource(GstElement*, GParamSpec*, gpointer)
PHONON-GST               END__: static void Phonon::Gstreamer::Pipeline::cb_setupSource(GstElement*, GParamSpec*, gpointer) [Took: 0s]
PHONON-GST             END__: GstStateChangeReturn Phonon::Gstreamer::Pipeline::setState(GstState) [Took: 0s]                                                                                                       
PHONON-GST           END__: void Phonon::Gstreamer::MediaObject::requestState(Phonon::State) [Took: 0.001s]                                                                                                         
PHONON-GST         END__: virtual void Phonon::Gstreamer::MediaObject::play() [Took: 0.001s]
PHONON-GST         Stream changed to 
PHONON-GST         BEGIN: void Phonon::Gstreamer::MediaObject::handleStreamChange()
PHONON-GST           false
PHONON-GST         END__: void Phonon::Gstreamer::MediaObject::handleStreamChange() [Took: 0s]
PHONON-GST         BEGIN: void Phonon::Gstreamer::MediaObject::handleAboutToFinish()
PHONON-GST           About to finish
PHONON-GST           total time -1
PHONON-GST           current time 0
PHONON-GST           remaining time -1
PHONON-GST           waiting for 0
PHONON-GST           [WARNING] aboutToFinishWait timed out!
PHONON-GST         END__: void Phonon::Gstreamer::MediaObject::handleAboutToFinish() [Took: 0s]
PHONON-GST         BEGIN: void Phonon::Gstreamer::MediaObject::handleAboutToFinish()
PHONON-GST           About to finish
PHONON-GST           total time -1
PHONON-GST           current time 0
PHONON-GST           remaining time -1
PHONON-GST           waiting for 0
PHONON-GST           [WARNING] aboutToFinishWait timed out!
PHONON-GST         END__: void Phonon::Gstreamer::MediaObject::handleAboutToFinish() [Took: 0s]
PHONON-GST         State change
PHONON-GST         BEGIN: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState)
PHONON-GST           Moving from "ready"  Phonon::StoppedState  to "paused"  Phonon::PausedState 
PHONON-GST         END__: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState) [Took: 0s]                                                                                                     
PHONON-GST         State change
PHONON-GST         BEGIN: void Phonon::Gstreamer::MediaObject::handleDurationChange(qint64)
PHONON-GST           2068
PHONON-GST         END__: void Phonon::Gstreamer::MediaObject::handleDurationChange(qint64) [Took: 0s]
PHONON-GST         BEGIN: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState)
PHONON-GST           Moving from "paused"  Phonon::PausedState  to "playing"  Phonon::PlayingState 
PHONON-GST         END__: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState) [Took: 0s]
PHONON-GST         BEGIN: void Phonon::Gstreamer::MediaObject::handleEndOfStream()
PHONON-GST           not skipping EOS
PHONON-GST           BEGIN: GstStateChangeReturn Phonon::Gstreamer::Pipeline::setState(GstState)
PHONON-GST             Transitioning to state "ready"
PHONON-GST             State change
PHONON-GST             BEGIN: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState)
PHONON-GST               Moving from "playing"  Phonon::PlayingState  to "paused"  Phonon::PausedState 
PHONON-GST             END__: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState) [Took: 0s]
PHONON-GST             State change
PHONON-GST             BEGIN: void Phonon::Gstreamer::MediaObject::handleDurationChange(qint64)
PHONON-GST               -1
PHONON-GST             END__: void Phonon::Gstreamer::MediaObject::handleDurationChange(qint64) [Took: 0s]
PHONON-GST             BEGIN: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState)
PHONON-GST               Moving from "paused"  Phonon::PausedState  to "ready"  Phonon::StoppedState 
PHONON-GST             END__: void Phonon::Gstreamer::MediaObject::handleStateChange(GstState, GstState) [Took: 0s]
PHONON-GST           END__: GstStateChangeReturn Phonon::Gstreamer::Pipeline::setState(GstState) [Took: 0.004s]
PHONON-GST         END__: void Phonon::Gstreamer::MediaObject::handleEndOfStream() [Took: 0.004s]
```

### Steps to reproduce
$ nix-env -i amarok-kf5
$ amarok
happens even after reboot.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.12, NixOS, 18.03pre131587.b6ddb9913f2 (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0`
 - channels(root): `"nixos-18.03pre131587.b6ddb9913f2"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by Anton-Latukha** [comment]
@cx405 Build normally now:
https://hydra.nixos.org/job/nixos/release-18.03/nixpkgs.amarok-kf5.x86_64-linux
https://hydra.nixos.org/job/nixpkgs/trunk/amarok-kf5.x86_64-linux

**Comment by orivej** [comment]
The issue does not say that it does not build, it says that it does not run. (I have not tested if it runs or not now.)

**Comment by mdevlamynck** [comment]
No, the issue is still present.

**Comment by mdevlamynck** [comment]
If it helps, after running `kbuildsycoca5` (as the user running amarok) amarok starts but some elements seem to be missing like the middle pannel.

**Comment by Anton-Latukha** [comment]
Then with `kbuildsycoca5` is a part of:
1. [plasma5: all menus gone after upgrade #41416](#41416)
1. [KDE will not automatically update it's start menu #40915](#40915)
1. [Application launcher menu is not updated #20776](#20776)
1. [plasma5: all menus gone after upgrade #41416](#41416)
1. [ksysguard desktop file does not have an Exec line #40916](#40916)

**Comment by lasers** [comment]
Please revisit this again. I do not get a dialog box saying "Amarok could not find any plugins" does it quit on me.

**Comment by SuperSandro2000** [comment]
Should be fixed with https://github.com/NixOS/nixpkgs/issues/10419.
